### PR TITLE
Release v0.2.0

### LINKED ISSUE #15: question: Does wasm-nn-ggml support streaming response like openai's API
### Summary

I have successfully tried to start llama-api-server.wasm application to work. However after I have tried to integrate with openAI compatible applications, I have found the following features:

1. llama-api-server does not support response as a stream, which commonly used with popular LLM applications.
2. Everytime I tried to make a request to llama-api-server, the server always tries to reload layers into GPU(nn-ggml-cuda support), which makes long latency, could this be optimized by loading layers into GPU only at startup?

This two features makes llama-utils less essential for serving llm applications. 
Could these shortages be fixed in the future?

### Appendix

_No response_

### LINKED ISSUE #25: bug: llama-api-server 不使用新的提示词
### Summary

 多次客户端访问使用不同提示词，结果却是一样的。参考如下输出：

% curl  -X POST http://0.0.0.0:8080/v1/chat/completions \
-H 'accept:application/json' \
-H 'Content-Type: application/json' \
-d '{"messages":[{"role":"system", "content":"You are a helpful AI assistant"}, {"role":"user", "content":"请计划一个南京3天的旅行"}], "model":"default"}'
{"id":"0c2c39ee-aaed-4380-9422-23daa90bfa99","object":"chat.completion","created":1702474374,"model":"default","choices":[{"index":0,"message":{"role":"assistant","content":"当然可以为您规划一个南京3天的旅行。以下是一个基本的旅行计划建议：\n\n**第1天：历史文化之旅**\n- 早上：参观中山陵，了解孙中山先生的生平和历史贡献。\n- 中午：在中山陵附近的餐厅享用午餐，品尝当地美食。\n- 下午：游览南京博物院，欣赏丰富的历史文物和艺术品。\n- 晚上：前往秦淮河夫子庙区域，体验夜游秦淮河，品尝夫子庙的小吃。\n\n**第2天：自然与现代结合**\n- 早上：前往紫金山天文台，了解天文知识，欣赏南京的自然风光。\n- 中午：在紫金山附近的餐厅用餐，享受山林间的宁静。\n- 下午：参观侵华日军南京大屠杀遇难同胞纪念馆，铭记历史。\n- 晚上：前往南京的新街口商业区，体验现代都市的繁华。\n\n**第3天：古迹与休闲**\n- 早上：参观明孝陵，感受明朝的皇家陵墓文化。\n- 中午：在明孝陵附近的餐厅享用午餐。\n- 下午：游览南京古城墙，如中华门城堡，体验古代防御工事。\n- 晚上：前往南京的鼓楼区，逛逛特色小店，享受悠闲的时光。\n\n请根据您的兴趣和实际情况调整行程。南京还有很多其他的景点和活动，如总统府、南京大学等，您可以根据自己的喜好添加或删减。祝您旅途愉快！"},"finish_reason":"stop"}],"usage":{"prompt_tokens":10,"completion_tokens":29,"total_tokens":39}}%                                                                                 (base) 

% curl  -X POST http://0.0.0.0:8080/v1/chat/completions \
-H 'accept:application/json' \
-H 'Content-Type: application/json' \
-d '{"messages":[{"role":"system", "content":"You are a helpful AI assistant"}, {"role":"user", "content":"请计划一个天津3天的旅行"}], "model":"default"}'
{"id":"d280487c-279c-44c4-9826-702043e4639b","object":"chat.completion","created":1702474434,"model":"default","choices":[{"index":0,"message":{"role":"assistant","content":"当然可以为您计划一个3天的旅行。请告诉我您的旅行计划建议：\n\n**第1天：历史文化之旅**\n- 早上：参观南京博物院，了解中国近代历史和文化遗产。\n- 中午：在中山陵附近的餐厅享用午餐，品尝当地特色美食。\n- 下午：游览明孝陵和紫金山天文台，感受明朝的文化遗产。\n- 可以在夫人的夫子庙区域，体验传统的江南水乡风情，品尝小吃。\n\n**第2天：自然与现代结合**\n- 早上：前往紫金山天文台，参观天文望远镜，感受自然之美。\n- 中午：在紫金山附近的餐厅用餐，享受山林间的宁静氛围。\n- 下午：参观南京大屠杀遇难者纪念馆，铭记历史，缅怀先烈。\n- 晚上：前往南京的新街口区，体验现代都市的繁华，品尝各种美食。\n\n第3天：休闲与购物**\n- 早上：参观南京古城墙，感受明朝的皇家建筑风格。\n- 中午：在明孝陵附近的餐厅用餐，享受一个轻松的午餐。\n- 前往南京的某个段，如门或玄武门，感受历史。\n- 晚上：在南京的夫子庙区域，体验当地的夜市，购买纪念品。\n\n请根据您的兴趣和实际情况调整行程，祝您在旅行景点和活动，如南京大学、灵谷寺等，您可以根据自己的喜好添加或删减。"},"finish_reason":"stop"}],"usage":{"prompt_tokens":10,"completion_tokens":29,"total_tokens":39}}%    

### Reproduction steps

发出两次请求，含有不同的提示词。
观察输出结果是第一次提示词的。

### Screenshots

DESCRIPTION

### Any logs you want to share for showing the specific issue

_No response_

### Model Information

yi-34b-chat

### Operating system information

Mac

### ARCH

arm64

### CPU Information

m2

### Memory Size

192G

### GPU Information

m2

### VRAM Size

192

### LINKED ISSUE #26: bug: api response content is ▅▅▅▅▅▅▅▅▅
### Summary

the api response content is error

`{"id":"060ff2a8-d603-4fd4-b772-6de8a72e51e4","object":"chat.completion","created":1702988129,"model":"Dolphin-2.0-Mistral-7B","choices":[{"index":0,"message":{"role":"assistant","content":"▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅"},"finish_reason":"stop"}],"usage":{"prompt_tokens":12,"completion_tokens":1,"total_tokens":13}}%`

### Reproduction steps

curl -X POST http://localhost:8080/v1/chat/completions -H 'accept:application/json' -H 'Content-Type: application/json' -d '{"messages":[{"role":"system", "content": "You are a helpful assistant."}, {"role":"user", "content": "Who is Robert Oppenheimer?"}], "model":"Dolphin-2.0-Mistral-7B"}'

### Screenshots

image

### Any logs you want to share for showing the specific issue

_No response_

### Model Information

dolphin-2.0-mistral-7b-ggml-model-q4_0.gguf

### Operating system information

macOS 14.2

### ARCH

arm64

### CPU Information

Apple M3 Max

### Memory Size

128G

### GPU Information

Apple M3

### VRAM Size

128G

### LINKED ISSUE #27: Replace the download link in `run-llm.sh`
### Summary

Replace the download link with the redirected links created by flows.network, so we can track the number for downloads.

### Appendix

_No response_

### PR DESCRIPTION
**Pull request by apepkuss**
- Release `llama-api-server v0.2.0` and `llama-chat v0.2.0`
- Fix #25 
- Fix #26 
- Fix #27 
- Fix #15

### CODE REVIEW

File: run-llm.sh [hunk 1]
```diff
@@ -99,37 +99,37 @@ download_model() {
...
+WizardLM-13B-V1.0-Uncensored::https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/wizardlm-13b-v1.0-uncensored.Q5_K_M.gguf
+WizardCoder-Python-7B-V1.0::https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/WizardCoder-Python-7B-V1.0-ggml-model-q4_0.gguf
+Zephyr-7B-Alpha::https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/zephyr-7b-alpha.Q5_K_M.gguf
+Orca-2-13B::https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/Orca-2-13b-ggml-model-q4_0.gguf
+Neural-Chat-7B-v3-1::https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/neural-chat-7b-v3-1-ggml-model-q4_0.gguf
+Starling-LM-7B-alpha::https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/starling-lm-7b-alpha.Q5_K_M.gguf
+Calm2-7B-Chat::https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/calm2-7b-chat.Q4_K_M.gguf
+Deepseek-Coder-6.7B::https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/deepseek-coder-6.7b-instruct.Q5_K_M.gguf
+Deepseek-LLM-7B-Chat::https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/deepseek-llm-7b-chat.Q5_K_M.gguf
 '
```
Comment (alabulei1) [inline run-llm.sh:134]: Please also replace the download link of wasm files.

https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/llama-api-server.wasm

https://code.flows.network/webhook/iwYN1SdN3AmPgR5ao5Gt/llama-chat.wasm

**Comment by alabulei1** [comment]
I  think the new WasmEdge ggml plugin also solved #21. So we will need to update the run-llm.sh again based on the comment here: https://github.com/second-state/llama-utils/issues/21#issuecomment-1851465074

And the temporary solution for #21 is here: https://github.com/second-state/llama-utils/commit/26159695edb5c7a76fb1f41e0bfbe07e4176c0d7

File: run-llm.sh (hunk 1 above)
Comment (alabulei1) [reply run-llm.sh:134]: I have verified the links above. It works well.

Comment (apepkuss) [reply run-llm.sh:134]: Accepted.

**Review by alabulei1 — APPROVED** [review]

**Pull request merged**